docs/rfcs/0004-rfc-captive-portal-port-53.md

```diff
@@ -0,0 +1,39 @@
+- Status: Pending
+- Date: 2023-09-29
+- Author: [@gela](https://github.com/gela)
+- RFC PR: TBD
+- Implementation GitHub issue: TBD
+
+
+## Summary
+This RFC documents discussions and proposed actions pertaining to the management of captive portals in the Mozilla VPN. The primary objective is to address security concerns and optimize user experience when they encounter a captive portal.
+
+## Motivation
+
+This RFC outlines the feasibility of revising the current captive portal mechanism. The captive portal IP is obtained from Guardian before the tunnel is established and is excluded from the tunnel. Additionally the VPN currently undergoes periodic captive portal checks, even when users are not authenticated. 
+
+Our current approach is inefficient and leads to potential security vulnerabilities such as [VPN-5007](https://mozilla-hub.atlassian.net/browse/VPN-5007). In regions where state and nation wide surveillance is common, organizations can setup an address which is the same as the captive portal DNS and monitor the traffic. These organizations can then fish the user into going to their deceptive URL that bypasses captive portal. While the user will eventually be redirected back to the captive portal landing page, an attacker can still verify traffic and respond with a malicious payload that can still compromise user security.
+
+The key considerations are as follows:
+
+- **Conditional Captive Portal Checks:** Captive portal checks should be performed when the user activates or deactivates VPN.
+- **Elimination of Captive Portal Checks:** If upon activation we detect a captive portal, fallback to port 53 to try and bypass the captive portal. This may permit users to bypass captive portals without sharing personal information.
```

Comment:
Could this provide more details about how falling back to port 53/DNS server may bypass the captive portal, and how this will be done?